internal/net/grpc/client.go

```diff
@@ -1105,27 +1144,44 @@ func (g *gRPCClient) Close(ctx context.Context) (err error) {
 	if g.stopMonitor != nil {
 		g.stopMonitor()
 	}
-	g.rangeConns(ctx, func(addr string, p pool.Conn) bool {
+	for _, addr := range g.ConnectedAddrs(ctx) {
 		derr := g.Disconnect(ctx, addr)
 		if errors.IsNot(derr, errors.ErrGRPCClientConnNotFound(addr)) {
 			err = errors.Join(err, derr)
 		}
-		return true
-	})
+	}
 	return err
 }
 
-func (g *gRPCClient) rangeConns(ctx context.Context, fn func(addr string, p pool.Conn) bool) error {
+func (g *gRPCClient) rangeConns(
+	ctx context.Context, action string, fn func(ctx context.Context, addr string, p pool.Conn) bool,
+) error {
```

Comment:
Just question: when can this `gRPCClient` be `nil`?

Reply:
Usually there is **no way** `gRPCClient` to be nil when you use a construction method like _client.New_, but in test code or some other way to initialize `g` might be `nil`.